{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Problem: Plasmid Design\n",
    "\n",
    "<hr>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Problem 1: Ups, our lab has no more BamHI\n",
    "\n",
    "Imagine, after the lecture you go in the lab to create the \"Hello world\" plasmid.\n",
    "Unfortunately, you find out that there is no more of the restriction enzyme BamHI - but you have the widely used EcoRI.\n",
    "\n",
    "**a)** Adapt the lecture's code for EcoRI.\n",
    "\n",
    "<details class=\"spoiler\"><summary>Show hint</summary>\n",
    "<div>\n",
    "Essentially replace BamHI with EcoRI. Just be careful about the overhangs and check if some code parts use position constants that need to be changed.\n",
    "</div></details>\n",
    "\n",
    "**b)** Check if there are any problems with this.\n",
    "\n",
    "<details class=\"spoiler\"><summary>Show hint</summary>\n",
    "<div>\n",
    "Does it cut only once within the MCS?\n",
    "</div></details>\n",
    "\n",
    "**c)** What is the insert fragment? Did it change?\n",
    "\n",
    "<details class=\"spoiler\"><summary>Show hint</summary>\n",
    "<div>\n",
    "Think about the homology arms.\n",
    "</div></details>\n",
    "\n",
    "**d)** Run the Gibson simulation as in the lecture and show the final construct? Do you see any reportable changes?\n",
    "\n",
    "<details class=\"spoiler\"><summary>Show hint</summary>\n",
    "<div>\n",
    "Should be almost for free if you adapted the code in (a). Just some constants for plotting may need a check. Otherwise, what about the distance between the promoter and the RBS. Is it larger? Check if there are now even more potential start codons.\n",
    "</div></details>\n",
    "\n",
    "\n",
    "# Problem 2: Linearize Backbone with PCR\n",
    "\n",
    "In the lecture we saw that there is a little hack in our code that may need polishing for a clean design: we just let the original RBS sequence\n",
    "with the original start codon and initial part of the protein in the construct and simply added an RBS inside the original CDS.\n",
    "That will lead to some Ribosomes starting at a wrong position.\n",
    "\n",
    "Remove this hack by a cleaner way to linearize the backbone, at the same time cutting out the RBS with the initial CDS fragment.\n",
    "We will do this via designing a forward and a backward primer and PCRing the backbone.\n",
    "\n",
    "**a)** Determine the sequence that you want to remove from the backbone.\n",
    "\n",
    "<details class=\"spoiler\"><summary>Show hint</summary>\n",
    "<div>\n",
    "Look for a Shine-Dalgarno-close sequence between the lac operator and the original start codon. This will be a good start for the cut. The end of the cut is a bit more free and can be chosen anywhere in the lacZ-alpha CDS.\n",
    "</div></details>\n",
    "\n",
    "**b)** Design primers with `primer3`\n",
    "\n",
    "<details class=\"spoiler\"><summary>Show hint</summary>\n",
    "<div>\n",
    "Be careful how they are arranged. You want to amplify the larger part between them, not the short insert.\n",
    "</div></details>\n",
    "\n",
    "**c)** Adapt the lecture's code to the new linearized backbone and run a simulated Gibson Assembly.\n",
    "\n",
    "<details class=\"spoiler\"><summary>Show hint</summary>\n",
    "<div>\n",
    "This is much like the previous exercise.\n",
    "</div></details>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "**License**: © 2025 Matthias Függer and Thomas Nowak. Licensed under [CC BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/)."
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
